doc/conf.py

```diff
@@ -286,6 +287,34 @@ def download(path, filename):
 # (source start file, name, description, authors, manual section).
 man_pages = [("index", "refl1d", program_title, ["Paul Kienzle"], 1)]
 
+# -- header for Jupyter notebooks --------------------------------------------
+# This is processed by Jinja2 and inserted before each notebook
+nbsphinx_prolog = r"""
+{% set docname = 'doc/' + env.doc2path(env.docname, base=None)|string %}
+
+.. raw:: html
+
+    <div class="admonition note">
+      This page was generated from
+      <a class="reference external" href="https://github.com/reflectometry/refl1d/tree/main/{{ docname|e }}">{{ docname|e }}</a>.
```

Comment:
`.../refl1d/tree/master/...` ?

Also, a couple of lines down uses `.../refl1d/blob/main/...` for the colab link.

Reply:
Checking, the `tree` gets replaced by `blob` when I click on the link.

The link obviously doesn't work since the notebook is only in a branch. Maybe we should use the git hash instead of the branch name so that it looks at the notebook at the time the docs were built? Then it doesn't matter if it is master or main and it doesn't matter if the docs were built in a branch.

Reply:
This was merged last week, so you should be able to try it out on the master branch.

I agree, if there's some way to use the commit or tag instead that would be useful.